// @wechatferry/core/dist/proto/room-data.cjs
'use strict';

const pb_1 = require('google-protobuf');

function _interopNamespaceCompat(e) {
    if (e && typeof e === 'object' && 'default' in e) return e;
    const n = Object.create(null);
    if (e) {
        for (const k in e) {
            n[k] = e[k];
        }
    }
    n.default = e;
    return n;
}

const pb_1__namespace = /*#__PURE__*/_interopNamespaceCompat(pb_1);

var __accessCheck = (obj, member, msg) => {
  if (!member.has(obj))
    throw TypeError("Cannot " + msg);
};
var __privateGet = (obj, member, getter) => {
  __accessCheck(obj, member, "read from private field");
  return getter ? getter.call(obj) : member.get(obj);
};
var __privateAdd = (obj, member, value) => {
  if (member.has(obj))
    throw TypeError("Cannot add the same private member more than once");
  member instanceof WeakSet ? member.add(obj) : member.set(obj, value);
};
var _one_of_decls;
const _RoomData = class _RoomData extends pb_1__namespace.Message {
  constructor(data) {
    super();
    __privateAdd(this, _one_of_decls, []);
    pb_1__namespace.Message.initialize(this, Array.isArray(data) ? data : [], 0, -1, [1], __privateGet(this, _one_of_decls));
    if (!Array.isArray(data) && typeof data == "object") {
      if ("members" in data && data.members != void 0) {
        this.members = data.members;
      }
      if ("field_2" in data && data.field_2 != void 0) {
        this.field_2 = data.field_2;
      }
      if ("field_3" in data && data.field_3 != void 0) {
        this.field_3 = data.field_3;
      }
      if ("field_4" in data && data.field_4 != void 0) {
        this.field_4 = data.field_4;
      }
      if ("room_capacity" in data && data.room_capacity != void 0) {
        this.room_capacity = data.room_capacity;
      }
      if ("field_7" in data && data.field_7 != void 0) {
        this.field_7 = data.field_7;
      }
      if ("field_8" in data && data.field_8 != void 0) {
        this.field_8 = data.field_8;
      }
    }
  }
  get members() {
    return pb_1__namespace.Message.getRepeatedWrapperField(this, _RoomData.RoomMember, 1);
  }
  set members(value) {
    pb_1__namespace.Message.setRepeatedWrapperField(this, 1, value);
  }
  get field_2() {
    return pb_1__namespace.Message.getFieldWithDefault(this, 2, 0);
  }
  set field_2(value) {
    pb_1__namespace.Message.setField(this, 2, value);
  }
  get field_3() {
    return pb_1__namespace.Message.getFieldWithDefault(this, 3, 0);
  }
  set field_3(value) {
    pb_1__namespace.Message.setField(this, 3, value);
  }
  get field_4() {
    return pb_1__namespace.Message.getFieldWithDefault(this, 4, 0);
  }
  set field_4(value) {
    pb_1__namespace.Message.setField(this, 4, value);
  }
  get room_capacity() {
    return pb_1__namespace.Message.getFieldWithDefault(this, 5, 0);
  }
  set room_capacity(value) {
    pb_1__namespace.Message.setField(this, 5, value);
  }
  get field_7() {
    return pb_1__namespace.Message.getFieldWithDefault(this, 7, "0");
  }
  set field_7(value) {
    pb_1__namespace.Message.setField(this, 7, value);
  }
  get field_8() {
    return pb_1__namespace.Message.getFieldWithDefault(this, 8, "0");
  }
  set field_8(value) {
    pb_1__namespace.Message.setField(this, 8, value);
  }
  static fromObject(data) {
    const message = new _RoomData({});
    if (data.members != null) {
      message.members = data.members.map((item) => _RoomData.RoomMember.fromObject(item));
    }
    if (data.field_2 != null) {
      message.field_2 = data.field_2;
    }
    if (data.field_3 != null) {
      message.field_3 = data.field_3;
    }
    if (data.field_4 != null) {
      message.field_4 = data.field_4;
    }
    if (data.room_capacity != null) {
      message.room_capacity = data.room_capacity;
    }
    if (data.field_7 != null) {
      message.field_7 = data.field_7;
    }
    if (data.field_8 != null) {
      message.field_8 = data.field_8;
    }
    return message;
  }
  toObject() {
    const data = {};
    if (this.members != null) {
      data.members = this.members.map((item) => item.toObject());
    }
    if (this.field_2 != null) {
      data.field_2 = this.field_2;
    }
    if (this.field_3 != null) {
      data.field_3 = this.field_3;
    }
    if (this.field_4 != null) {
      data.field_4 = this.field_4;
    }
    if (this.room_capacity != null) {
      data.room_capacity = this.room_capacity;
    }
    if (this.field_7 != null) {
      data.field_7 = this.field_7;
    }
    if (this.field_8 != null) {
      data.field_8 = this.field_8;
    }
    return data;
  }
  serialize(w) {
    const writer = w || new pb_1__namespace.BinaryWriter();
    if (this.members.length)
      writer.writeRepeatedMessage(1, this.members, (item) => item.serialize(writer));
    if (this.field_2 != 0)
      writer.writeInt32(2, this.field_2);
    if (this.field_3 != 0)
      writer.writeInt32(3, this.field_3);
    if (this.field_4 != 0)
      writer.writeInt32(4, this.field_4);
    if (this.room_capacity != 0)
      writer.writeInt32(5, this.room_capacity);
    if (this.field_7 != "0")
      writer.writeInt64String(7, this.field_7);
    if (this.field_8 != "0")
      writer.writeInt64String(8, this.field_8);
    if (!w)
      return writer.getResultBuffer();
  }
  static deserialize(bytes) {
    const reader = bytes instanceof pb_1__namespace.BinaryReader ? bytes : new pb_1__namespace.BinaryReader(bytes), message = new _RoomData();
    while (reader.nextField()) {
      if (reader.isEndGroup())
        break;
      switch (reader.getFieldNumber()) {
        case 1:
          reader.readMessage(message.members, () => pb_1__namespace.Message.addToRepeatedWrapperField(message, 1, _RoomData.RoomMember.deserialize(reader), _RoomData.RoomMember));
          break;
        case 2:
          message.field_2 = reader.readInt32();
          break;
        case 3:
          message.field_3 = reader.readInt32();
          break;
        case 4:
          message.field_4 = reader.readInt32();
          break;
        case 5:
          message.room_capacity = reader.readInt32();
          break;
        case 7:
          message.field_7 = reader.readInt64String();
          break;
        case 8:
          message.field_8 = reader.readInt64String();
          break;
        default:
          reader.skipField();
      }
    }
    return message;
  }
  serializeBinary() {
    return this.serialize();
  }
  static deserializeBinary(bytes) {
    return _RoomData.deserialize(bytes);
  }
};
_one_of_decls = new WeakMap();
exports.RoomData = _RoomData;
((RoomData2) => {
  var _one_of_decls2;
  const _RoomMember = class _RoomMember extends pb_1__namespace.Message {
    constructor(data) {
      super();
      __privateAdd(this, _one_of_decls2, []);
      pb_1__namespace.Message.initialize(this, Array.isArray(data) ? data : [], 0, -1, [], __privateGet(this, _one_of_decls2));
      if (!Array.isArray(data) && typeof data == "object") {
        if ("wxid" in data && data.wxid != void 0) {
          this.wxid = data.wxid;
        }
        if ("name" in data && data.name != void 0) {
          this.name = data.name;
        }
        if ("state" in data && data.state != void 0) {
          this.state = data.state;
        }
      }
    }
    get wxid() {
      return pb_1__namespace.Message.getFieldWithDefault(this, 1, "");
    }
    set wxid(value) {
      pb_1__namespace.Message.setField(this, 1, value);
    }
    get name() {
      return pb_1__namespace.Message.getFieldWithDefault(this, 2, "");
    }
    set name(value) {
      pb_1__namespace.Message.setField(this, 2, value);
    }
    get state() {
      return pb_1__namespace.Message.getFieldWithDefault(this, 3, 0);
    }
    set state(value) {
      pb_1__namespace.Message.setField(this, 3, value);
    }
    static fromObject(data) {
      const message = new _RoomMember({});
      if (data.wxid != null) {
        message.wxid = data.wxid;
      }
      if (data.name != null) {
        message.name = data.name;
      }
      if (data.state != null) {
        message.state = data.state;
      }
      return message;
    }
    toObject() {
      const data = {};
      if (this.wxid != null) {
        data.wxid = this.wxid;
      }
      if (this.name != null) {
        data.name = this.name;
      }
      if (this.state != null) {
        data.state = this.state;
      }
      return data;
    }
    serialize(w) {
      const writer = w || new pb_1__namespace.BinaryWriter();
      if (this.wxid.length)
        writer.writeString(1, this.wxid);
      if (this.name.length)
        writer.writeString(2, this.name);
      if (this.state != 0)
        writer.writeInt32(3, this.state);
      if (!w)
        return writer.getResultBuffer();
    }
    static deserialize(bytes) {
      const reader = bytes instanceof pb_1__namespace.BinaryReader ? bytes : new pb_1__namespace.BinaryReader(bytes), message = new _RoomMember();
      while (reader.nextField()) {
        if (reader.isEndGroup())
          break;
        switch (reader.getFieldNumber()) {
          case 1:
            message.wxid = reader.readString();
            break;
          case 2:
            message.name = reader.readString();
            break;
          case 3:
            message.state = reader.readInt32();
            break;
          default:
            reader.skipField();
        }
      }
      return message;
    }
    serializeBinary() {
      return this.serialize();
    }
    static deserializeBinary(bytes) {
      return _RoomMember.deserialize(bytes);
    }
  };
  _one_of_decls2 = new WeakMap();
  RoomData2.RoomMember = _RoomMember;
})(exports.RoomData || (exports.RoomData = {}));
