// @wechatferry/core/dist/types.mjs
var WechatAppMessageType = /* @__PURE__ */ ((WechatAppMessageType2) => {
  WechatAppMessageType2[WechatAppMessageType2["Text"] = 1] = "Text";
  WechatAppMessageType2[WechatAppMessageType2["Img"] = 2] = "Img";
  WechatAppMessageType2[WechatAppMessageType2["Audio"] = 3] = "Audio";
  WechatAppMessageType2[WechatAppMessageType2["Video"] = 4] = "Video";
  WechatAppMessageType2[WechatAppMessageType2["Url"] = 5] = "Url";
  WechatAppMessageType2[WechatAppMessageType2["Attach"] = 6] = "Attach";
  WechatAppMessageType2[WechatAppMessageType2["Open"] = 7] = "Open";
  WechatAppMessageType2[WechatAppMessageType2["Emoji"] = 8] = "Emoji";
  WechatAppMessageType2[WechatAppMessageType2["VoiceRemind"] = 9] = "VoiceRemind";
  WechatAppMessageType2[WechatAppMessageType2["ScanGood"] = 10] = "ScanGood";
  WechatAppMessageType2[WechatAppMessageType2["Good"] = 13] = "Good";
  WechatAppMessageType2[WechatAppMessageType2["Emotion"] = 15] = "Emotion";
  WechatAppMessageType2[WechatAppMessageType2["CardTicket"] = 16] = "CardTicket";
  WechatAppMessageType2[WechatAppMessageType2["RealtimeShareLocation"] = 17] = "RealtimeShareLocation";
  WechatAppMessageType2[WechatAppMessageType2["ChatHistory"] = 19] = "ChatHistory";
  WechatAppMessageType2[WechatAppMessageType2["MiniProgram"] = 33] = "MiniProgram";
  WechatAppMessageType2[WechatAppMessageType2["MiniProgramApp"] = 36] = "MiniProgramApp";
  WechatAppMessageType2[WechatAppMessageType2["Channels"] = 51] = "Channels";
  WechatAppMessageType2[WechatAppMessageType2["GroupNote"] = 53] = "GroupNote";
  WechatAppMessageType2[WechatAppMessageType2["ReferMsg"] = 57] = "ReferMsg";
  WechatAppMessageType2[WechatAppMessageType2["Transfers"] = 2e3] = "Transfers";
  WechatAppMessageType2[WechatAppMessageType2["RedEnvelopes"] = 2001] = "RedEnvelopes";
  WechatAppMessageType2[WechatAppMessageType2["ReaderType"] = 100001] = "ReaderType";
  return WechatAppMessageType2;
})(WechatAppMessageType || {});
var WechatMessageType = /* @__PURE__ */ ((WechatMessageType2) => {
  WechatMessageType2[WechatMessageType2["Moment"] = 0] = "Moment";
  WechatMessageType2[WechatMessageType2["Text"] = 1] = "Text";
  WechatMessageType2[WechatMessageType2["Image"] = 3] = "Image";
  WechatMessageType2[WechatMessageType2["Voice"] = 34] = "Voice";
  WechatMessageType2[WechatMessageType2["VerifyMsg"] = 37] = "VerifyMsg";
  WechatMessageType2[WechatMessageType2["PossibleFriendMsg"] = 40] = "PossibleFriendMsg";
  WechatMessageType2[WechatMessageType2["ShareCard"] = 42] = "ShareCard";
  WechatMessageType2[WechatMessageType2["Video"] = 43] = "Video";
  WechatMessageType2[WechatMessageType2["Emoticon"] = 47] = "Emoticon";
  WechatMessageType2[WechatMessageType2["Location"] = 48] = "Location";
  WechatMessageType2[WechatMessageType2["App"] = 49] = "App";
  WechatMessageType2[WechatMessageType2["VoipMsg"] = 50] = "VoipMsg";
  WechatMessageType2[WechatMessageType2["StatusNotify"] = 51] = "StatusNotify";
  WechatMessageType2[WechatMessageType2["VoipNotify"] = 52] = "VoipNotify";
  WechatMessageType2[WechatMessageType2["VoipInvite"] = 53] = "VoipInvite";
  WechatMessageType2[WechatMessageType2["MicroVideo"] = 62] = "MicroVideo";
  WechatMessageType2[WechatMessageType2["VerifyMsgEnterprise"] = 65] = "VerifyMsgEnterprise";
  WechatMessageType2[WechatMessageType2["Transfer"] = 2e3] = "Transfer";
  WechatMessageType2[WechatMessageType2["RedEnvelope"] = 2001] = "RedEnvelope";
  WechatMessageType2[WechatMessageType2["MiniProgram"] = 2002] = "MiniProgram";
  WechatMessageType2[WechatMessageType2["GroupInvite"] = 2003] = "GroupInvite";
  WechatMessageType2[WechatMessageType2["File"] = 2004] = "File";
  WechatMessageType2[WechatMessageType2["SysNotice"] = 9999] = "SysNotice";
  WechatMessageType2[WechatMessageType2["Sys"] = 1e4] = "Sys";
  WechatMessageType2[WechatMessageType2["Recalled"] = 10002] = "Recalled";
  return WechatMessageType2;
})(WechatMessageType || {});

export { WechatAppMessageType, WechatMessageType };
